### PR TITLE
Ashlanders: Phase 3

### PR DESCRIPTION
**Pull request by Captain277**
## About The Pull Request

**Mobs:**
1. **Adds the Shank.**
2. **Adds the Stormdrifter.**
3. **Goliaths can now be milked for gunpowder.**

**Structures:**
1. **Adds Ashlander Structures: Magma Forge and Brick Press.**
2. **Adds Ashlander Catwalks.**
3. **Adds Goliath Hide curtains.**
4. **Adds a Tying Post.**

**Map Edits:**
1. **Removes the Harmony colony.**
2. **Removes Ashlander Camp.**
3. **Adds Ashlander Village.**

**Adjustments/Changes:**
1. **Tweaks Bone Axes and Spears.**
2. **Restricts Tribal Recipes.**
3. **Rearranges Ashlander structures and icons into bespoke files.**
4. **Beakers can now scoop ash piles off the ground.**
5. **Shows Reagents in Stone Mortars on Examine.**
6. **Creates Preset Types for Bone Tables/Benches & Sandstone Tables/Benches.**
7. **Wood Fences are now craftable with planks.**
8. **Boosts Ashlander Darksight.**
9. **Changes the lid colors of Ashlander containers.**
10. **Gives Ashlanders a Dedicated Language.**

**Alchemy:**
1. **Adds a craftable Ashlander Alchemy Station.**
2. **Adds a craftable stone mortar.**
3. **Creates New Reagent: "Alchemical Base"**
4. **Adds a Fertilizer recipe.**
5. **Adds an Alternate Tallow recipe.**
6. **Adds a Soap recipe.**
7. **Adds a Charcoal Stick recipe.**
8. **Adds a craftable Calcinator.**
9. **Adds the Industrial Plant Cersut.**
10. **Creates a craftable Stone Dropper.**
11. **Adds Juhtak Poultice recipe.**
12. **Adds Pyrrhlea Poultice recipe.**

**Craftables/Content:**
1. **Creates Craftable Saddles.**
2. **Adds Sandstone Flooring.**
3. **Adds Wooden Roofing.**
4. **Adds the craftable lava-proof boat and oar.**
5. **Adds elder stones and a harvestable outcrop.**
6. **Adds Charcoal Sticks - a pen subtype.**
7. **Adds craftable Sandstone jars.**
8. **Adds craftable Cooking Spit.**
9. **Adds primitive Soap.**
10. **Adds Goliath hide gloves.**
11. **Adds Goliath hide ore bags.**
12. **Adds bone specific arrow recipes for Ashlanders.**
13. **Provides specialized armor for Ashlanders.**
14. **Adds Bone Talisman (Disenchanted) to Loadout.**
15. **Adds Goliath half-cloak.**
18. **Adds Goliath plant bag.**

**Weaponry:**
1. **Adds the Stun Lance.**
2. **Removes Firing Pins from Muskets.**
3. **Adds the Bone Sword and Elder Bone Sword.**
4. **Adds Shiv Knife.**
5. **Removes Musket safety.**
6. **Removes the safety from bows.**

## Why It's Good For The Game

**Mobs:**
1. _Gutshanks can now be matured into Shanks, which serve as mounts for Ashlanders. If a Shank has a saddle on it, you can set its name/rename it with a charcoal stick._
2. _Stormdrifters are floating mobs who wander the Eastern Plains aimlessly, having been blown off-course by storms. They likely have some kind of utility._
3. _Goliaths eat lots of sulfur and saltpeter-rich rocks and sand, and now canonically create gunpowder as a waste byproduct. This can now be collected by Ashlanders to replenish musket reserves._

**Structures:**
1. _Ashlanders need a way to generate rods to create flooring for their structures, and also a way to access lead sheets to make musket balls. The Magma Forge creates rods when fed iron ore, lead ingots when fed lead ore, copper ingots when fed copper ore, and slag when fed sand. To encourage more thematic construction, Ashlanders can use a Brick Press to create sandstone bricks when they feed it sand._
2. _I wanted a bridge/catwalk system for Ashlanders, and this is what I came up with._
3. _Instead of heavy stone doors and stuff, I thought goliath hide curtains would be pretty nifty. So we've got those now!_
4. _This craftable bone tying post allows you to anchor your mount in one spot, just like in the Wild West!_

**Map Edits:**
1. _As much as I love Harmony, it has served its narrative purpose. Instead of extending Lavaland with another Z-Level, as I planned a while back, I'm going to instead remap East to be the new Ashlander spawn zone._
2. _The exile POI for Ashlanders no longer serves purpose._
3. _It is instead replaced by this village!_

**Adjustments/Changes:**
1. _Bone axes and spears now use their bespoke sprites, instead of material inheritance sprites._
2. _To craft Tribal recipes, you must now be an Ashlander. This should encourage more commercial trade between Ashlanders and station crew._
3. _I had to create a new file and icon sheet for Ashlander specific structures. I have moved the primitive seed press and the yurt to this file, and adjusted the relevant areas in code._
4. _I was surprised to discover this function didn't exist. It seems obvious that you should be able to sweep ash up. Most people don't even realize ash is a valid reagent type. Now it can be utilized somewhere, hopefully! Besides just me, I mean._
5. _There have been complaints about the confusion around stone mortar contents on Surt. The ability to see reagents in a beaker used to exist, but appears to have been removed somewhere. It was a pain, but I figured out how to restore it for stone mortars. Now Ashlanders can safely know whether they're about to drink water or gunpowder._
6. _I just wanted these so I could map them in to replace most of the wood in the Ashlander village._
7. _I've seen some pretty weird fencing solutions for when the Ashlander corral breaks. I'm debating a bone-based fence for it, but in lieu of that, these should be craftable anyways._
8. _Originally I had this set to 5. It turns out 5 was too low. Even 7 feels a little inadequate. I'm considering other solutions._
9. _To match with more bone and less wood, these should be changed!_
10. _Apparently I was using an Admin event language all along. Now they have a language of their own!_

**Alchemy:**
1. _I know there isn't like, a super obvious utility for this yet, but it feels like something that might be useful later. Maybe for advanced Ashlander medicines like poultices or something. Maybe a Phase 4 thing._
2. _The stone mortar functions as a beaker, and can be used to contain reagents for Ashlanders, such as gunpowder._
3. _This chemical serves as the core reagent in Ashlander alchemical recipes. It helps to prevent overlap in recipes considering the limited number of reagents Ashlanders have access to. It can be crafted by combining ash and sulphuric acid._
4. _Ashlander players have also requested access to fertilizer. Although you can grind up plants currently to provide plots with nutrients, giving the players access to a baseline fertilizer recipe (which I was surprised to see didn't exist/doesn't exist in a way Ashlanders can fairly access) works too. It can be crafted by combining equal parts tallow, ash, and alchemical base._
5. _Tallow is vital for a couple of my early concept alchemical formulas. It also serves to add additional utility to raw meat which might otherwise go unused. I designed this alternate recipe to require little more than grinding meat down into its components. It can be crafted by combining equal parts triglyceride, protein, and alchemical base._
6. _Butchering animals is dirty work. There have been multiple requests for a way to wash dirty items. Unfortunately, providing Ashlanders with a puddle of water or other water-based vectors would undermine the purpose of farming Gutshanks. However, primitive cultures knew about soap and how to create it. It can be crafted by combining equal parts tallow, water, and ash._
7. _That Charcoal has to be made somewhere! It can be crafted by combining equal parts tallow, ash, and alchemical base._
8. _In alchemy, Calcinators are used to reduce ingredients to ash and release their interior potential. This is a fancy way of saying you burn stuff into ash. This machine was a total headache to me, and took up the majority of my code work on this particular day. It's essentially a bastard child of the reagent grinder designed to burn wood or bone (only bone really matters for Ashlanders), and pump out ash for use in recipes._
9. _The eagle eyed may have noticed earlier that Alchemical Base requires sulphuric acid to craft. This is possible due to the addition of Cersut. This leafy plant leeches Sulphur out of the soil, which is processed through its body. Consequently, the leaves retain a significant amount of sulphuric acid. Cersut is not designed for any form of consumption. It is purely an industrial plant._
10. _I've discovered that due to some uneven production results, you can get some total nightmare situations with Ashlander beakers. I've considered adding some kind of chem master to allow for faster refinement, but I mean. Where's the fun in that?_
11. _This poultice can be used to heal brute damage in a more focused way. Made by mixing ten parts alchemy base with ten parts bicaridine._
12. _As the above, but used to heal burn damage. Made by mixing ten parts alchemy base and ten parts kelotane._

**Craftables/Content:**
1. _Saddles allow you to ride Shanks when crafted and applied._
2. _Now Ashlanders can construct interior floors out of sandstone._
3. _Now we can have wood roofs, instead of just steel ones. This is basically a fluff addition._
4. _The current redesign for Lavaland East has a lot of lava rivers and stuff going on. It'd be nice if Ashlanders had special boats to sail over it with._
5. _Older Lavaland recipes for certain tools required Cultist soul stones. Obviously on Main this makes sense, but less so here. So I've created Elder Stones, slightly anomalous gems that can be found in special outcrops. I've come up with some lore for these, but it doesn't belong here. They're used in recipes for things like lava-proof boats, Ashlander welders, and Elder Swords, all of which are meant to be rare._
6. _Originally I had this concept of Ashlanders using charcoal sticks to draw pictures and stuff, but I realized it would just lead to graffiti and English letters all over my sand. So instead I've made them primitive pens with no vector for paper, etc. They can be used to label stone mortars, and feature in at least one crafting recipe._
7. _Since there's no wood on Surt, wood buckets don't make sense. Now Ashlanders can craft sandstone jars - which function as buckets - in their stead._
8. _The people have cried out for a way to cook food. I've made this grill subtype that's just a spit over fire. Originally I was going to make it require charcoal sticks as fuel to consume, but I think that'd just add tedium (and I kinda don't wanna code it.)_
9. _It's ugly and it doesn't smell fancy, but it sure cleans stuff up._
10. _Fashion and utility! These gloves will protect you from stinging nettles when farming._
11. _Picking up ore one at a time is kinda lame at this point._
12. _Since there's no way for Ashlanders to reliably get wood, I've made it so they can craft arrows just using bone._
13. _Ashlanders have been using borrowed Gladiator uniforms for a while. Now they have Babylonian style lamellar armor._
14. _Disenchanted bone talismans may now be selected in the accessory loadout, at the request of players who have received them as gifts from Ashlanders._
15. _This is purely an aesthetic/style addition._
16. _Adds Plant Bags for Ashlanders._

**Weaponry:**
1. _Designed to work in tandem with jousting, Stun Lances are more of a tool for flavor._
2. _Muskets are ancient, primitive tech. They wouldn't use firing pins._
3. _As with the above, mechanical safeties didn't exist for decades/centuries after muskets._
4. _Bone swords are neat! Elder Bone Swords were a recommendation by someone, but I really loved the concept. Both are craftable, although the Elder requires advanced (for Ashlanders) materials._
5. _I'm already touching the misc weapons file and someone suggested a cheap shiv made of duct tape and knife blades. Sounded fun._
6. _Bows shouldn't have safeties._

## Changelog

:cl:
add: Adds the Shank.
add: Adds craftable saddles.
add: Adds craftable stone mortar.
add: Adds gunpowder milking feature to Goliaths.
add: Adds Stun Lance.
add: Adds Ashlander crafting structures (Magma Forge and Brick Press)
add: Adds Ashlander alchemy station.
add: Adds sandstone flooring.
add: Adds wooden roofing.
add: Adds the bone sword and elder bone sword.
add: Adds lava-proof boats.
add: Adds elder stones.
add: Adds Ashlander catwalks.
add: Adds Ashlander Village to Lavaland East.
add: Adds a shiv.
add: Adds Goliath hide curtains.
add: Adds alchemical system.
add: Adds alchemical recipes: alchemical base, fertilizer, tallow, soap, charcoal sticks.
add: Adds Calcinator.
add: Adds industrial plant 'Cersut'.
add: Adds primitive soap and charcoal stick pens.
add: Adds sandstone jars.
add: Adds cooking spit.
add: Adds stone dropper.
add: Adds preset bone & sandstone tables and benches.
del: Removes Harmony colony.
del: Removes Ashlander Exile POI.
tweak: Removes firing pin requirement for muskets.
tweak: Changes Bone Axe and Spear sprites.
tweak: Restricts Tribal recipes to Ashlanders.
tweak: Rearranges Ashlander structures into specialized files.
tweak: Allows inspection of reagents in stone mortars.
/:cl:

### CODE REVIEW

**Comment by silicons** [comment]
tag prs with [WIP] or yourself pls

**Comment by TheLordME** [comment]
grafik
this seems a bit inconsistant

**Comment by TheLordME** [comment]
grafik

**Comment by TheLordME** [comment]
`proc name: stack trace (/proc/_stack_trace)
usr: biteroankles/(Gedara)
usr.loc: (Lava Land (East) - Ashlander Village (116,19,22))
src: null
call stack:
stack trace("The bentar attempted to add a ...", "code/modules/reagents/chemistr...", 163)
/datum/reagents (/datum/reagents): add reagent("Dylovene", 2, /list (/list), 0)
the bentar (/obj/item/reagent_containers/food/snacks/grown): Initialize(0, "bentars")
Atoms (/datum/controller/subsystem/atoms): InitAtom(the bentar (/obj/item/reagent_containers/food/snacks/grown), /list (/list))
the bentar (/obj/item/reagent_containers/food/snacks/grown): New(0, "bentars")
bentars (/datum/seed/bentars): harvest(Gedara (/mob/living/carbon/human), 0, null, null)
the bentar (/obj/machinery/portable_atmospherics/hydroponics/soil): harvest(Gedara (/mob/living/carbon/human))
the bentar (/obj/machinery/portable_atmospherics/hydroponics/soil): attack hand(Gedara (/mob/living/carbon/human))
Gedara (/mob/living/carbon/human): UnarmedAttack(the bentar (/obj/machinery/portable_atmospherics/hydroponics/soil), 2)
Gedara (/mob/living/carbon/human): melee attack chain(the bentar (/obj/machinery/portable_atmospherics/hydroponics/soil), 2, /list (/list))
Gedara (/mob/living/carbon/human): ClickOn(the bentar (/obj/machinery/portable_atmospherics/hydroponics/soil), "icon-x=16;icon-y=10;left=1;but...", null)
the bentar (/obj/machinery/portable_atmospherics/hydroponics/soil): Click(the dirt (115,19,22) (/turf/simulated/floor/outdoors/dirt/lavaland/interior), "mapwindow.map", "icon-x=16;icon-y=10;left=1;but...")
BiterOAnkles (/client): Click(the bentar (/obj/machinery/portable_atmospherics/hydroponics/soil), the dirt (115,19,22) (/turf/simulated/floor/outdoors/dirt/lavaland/interior), "mapwindow.map", "icon-x=16;icon-y=10;left=1;but...")`

dylovene is refered to as antitoxin in code, which is causing the runtime a lot

**Comment by QRQ-2817** [comment]
-No fuel for bonfire, unless you get lucky and get towercaps via weeds
-Bonfire grills don't work - they straight up have no code to cook meat on them?
-No way to increase nutrients of farm plots? I thought you could compost stuff, but maybe that's only for hydro trays?
-Bonfires barely produce light, and are big - maybe some smaller lanterns of some kind for the village?

Loving this so far, can't wait to see a Phase 4

File: code/game/objects/items/weapons/material/foam.dm [hunk 1]
```diff
@@ -37,7 +37,7 @@
 	hitsound = 'sound/effects/bodyfall3.ogg'
 
 /obj/item/material/twohanded/spear/foam/Initialize(mapload, material_key)
-	return ..(mapload,"foam")
```
Comment (silicons) [inline code/game/objects/items/weapons/material/foam.dm:40]: no
this needs to **return**

File: code/game/objects/items/weapons/melee/misc.dm [hunk 2]
```diff
@@ -228,6 +233,173 @@
...
+	var/flame_intensity = 2
+	var/flame_color = "#FF9933"
+	var/SA_bonus_damage = 25 // 50 total against demons and aberrations.
+	var/SA_vulnerability = MOB_CLASS_DEMONIC | MOB_CLASS_ABERRATION
+
+/obj/item/melee/ashlander/elder/Initialize(mapload)
+	. = ..()
+	AddComponent(/datum/component/anti_magic, TRUE, TRUE, FALSE, null, null, FALSE)
+
+/obj/item/melee/ashlander/elder/afterattack(atom/A, mob/user as mob)
```
Comment (silicons) [inline code/game/objects/items/weapons/melee/misc.dm:264]: remember to not copypaste "as x" from things
as thing is used in verb arguments
if a proc isn't a verb you don't need it at all

File: code/game/atoms/atoms_movable.dm [hunk 3]
```diff
@@ -533,4 +533,10 @@
 	qdel(em_render)
 	em_render = null
 
+/**
+ * Checks if we can avoid things like landmine, lava, etc, whether beneficial or harmful.
+ */
+/atom/movable/proc/is_avoiding_ground()
+    return (movement_type != GROUND) || throwing
+
```
Comment (silicons) [inline code/game/atoms/atoms_movable.dm:541]: please shove this below the comment below it
the comment below it refers to the emissives stuff

File: code/game/objects/items/weapons/material/foam.dm (hunk 1 above)
Comment (Captain277) [reply code/game/objects/items/weapons/material/foam.dm:40]: That's odd. I didn't remove this line in my code. I'm looking over at it right now and it's still there.
image

Comment (Captain277) [reply code/game/objects/items/weapons/material/foam.dm:40]: Disregard. I'm not sure why the return was removed. I don't remember doing that. Easy fix.

File: code/game/objects/items/weapons/melee/misc.dm (hunk 2 above)
Comment (Captain277) [reply code/game/objects/items/weapons/melee/misc.dm:264]: Good catch. Removed.

File: code/game/atoms/atoms_movable.dm (hunk 3 above)
Comment (Captain277) [reply code/game/atoms/atoms_movable.dm:541]: Shifted down!

**Comment by Captain277** [comment]
> -No fuel for bonfire, unless you get lucky and get towercaps via weeds -Bonfire grills don't work - they straight up have no code to cook meat on them? -No way to increase nutrients of farm plots? I thought you could compost stuff, but maybe that's only for hydro trays? -EDIT: ^^^ Just grind up crops and pour the unseeable contents onto the plots. -Bonfires barely produce light, and are big - maybe some smaller lanterns of some kind for the village? -New armor is nice, but being able to craft or at least have a tunic would be nice too. Don't need armor when farming. -Not being able to see the contents of wooden buckets / stone mortars is hugely problematic. "Is this full of water, or gunpowder?"
> 
> Loving this so far, can't wait to see a Phase 4

Heyo! Thanks for all your feedback. You helped point out some pretty big QOL issues, like the whole "can't examine beakers" thing. I've decided that wood will be incredibly rare for Ashlanders, and probably ultimately removed from their ecosystem entirely.

**Comment by QRQ-2817** [comment]
Ashlander gameplay got me droolin for more. Some additional notes:

- A way to make bone hitching posts?
- No way to craft stone beakers, they're handy to take water on a journey, but only 2 available
- Bows have a safety toggle
- Arrows should be made with bone, if wood is being phased out
- Still definitely need a way to make outfits, or have spares available

**Comment by Captain277** [comment]
> Ashlander gameplay got me droolin for more. Some additional notes:
> 
> * A way to make bone hitching posts?
> * No way to craft stone beakers, they're handy to take water on a journey, but only 2 available
> * Bows have a safety toggle
> * Arrows should be made with bone, if wood is being phased out
> * Still definitely need a way to make outfits, or have spares available

Hitching posts will be craftable with the next commit.
Beakers are craftable, actually! Look for Stone Mortar on the crafting menu.
Good catch on the bow and arrow issues!
Outfits are a tricky one, honestly. So there are certain materials that I want to restrict giving to Ashlanders, and raw cloth is one of them. I've got issues with how cloth might be used by Ashlanders for things like bedsheets, etc. I'm not against giving them a way to make more tunics, I'm just not sure what the actual usefulness would be. You've requested it twice now, though, so I'd love to hear your thought process on it!

**Comment by QRQ-2817** [comment]
> > Ashlander gameplay got me droolin for more. Some additional notes:
> > 
> > * A way to make bone hitching posts?
> > * No way to craft stone beakers, they're handy to take water on a journey, but only 2 available
> > * Bows have a safety toggle
> > * Arrows should be made with bone, if wood is being phased out
> > * Still definitely need a way to make outfits, or have spares available
> 
> Hitching posts will be craftable with the next commit. Beakers are craftable, actually! Look for Stone Mortar on the crafting menu. Good catch on the bow and arrow issues! Outfits are a tricky one, honestly. So there are certain materials that I want to restrict giving to Ashlanders, and raw cloth is one of them. I've got issues with how cloth might be used by Ashlanders for things like bedsheets, etc. I'm not against giving them a way to make more tunics, I'm just not sure what the actual usefulness would be. You've requested it twice now, though, so I'd love to hear your thought process on it!

Unsure on how to make them, I just wanted an option to not spawn with armor like I'm some experienced warrior. 

Also, tribal-crafted items are supposed to be valuable to be tradable, right? Are the tools any different than a basic set? Could have them function quicker or something. Additionally, the primitive welder causes eye damage, while we have no eye protection. I attempted to make Imidazoline to fix my eyes, but can't grind up raw carbon, and hydrogen is unobtainable.

Also, minor thing, but masks of some kind would be cool.

Additional minor thing, naming your shank would be sick as fuck. Can't wait to see their final form. More insane feature idea: I wonder if you could use the cargo trolley linking system to "hook up" a primitive sled to a shank, have it haul your trade goods around.

**Comment by QRQ-2817** [comment]
> > > Ashlander gameplay got me droolin for more. Some additional notes:
> > > 
> > > * A way to make bone hitching posts?
> > > * No way to craft stone beakers, they're handy to take water on a journey, but only 2 available
> > > * Bows have a safety toggle
> > > * Arrows should be made with bone, if wood is being phased out
> > > * Still definitely need a way to make outfits, or have spares available
> > 
> > 
> > Hitching posts will be craftable with the next commit. Beakers are craftable, actually! Look for Stone Mortar on the crafting menu. Good catch on the bow and arrow issues! Outfits are a tricky one, honestly. So there are certain materials that I want to restrict giving to Ashlanders, and raw cloth is one of them. I've got issues with how cloth might be used by Ashlanders for things like bedsheets, etc. I'm not against giving them a way to make more tunics, I'm just not sure what the actual usefulness would be. You've requested it twice now, though, so I'd love to hear your thought process on it!
> 
> Unsure on how to make them, I just wanted an option to not spawn with armor like I'm some experienced warrior.
> 
> Also, tribal-crafted items are supposed to be valuable to be tradable, right? Are the tools any different than a basic set? Could have them function quicker or something. Additionally, the primitive welder causes eye damage, while we have no eye protection. I attempted to make Imidazoline to fix my eyes, but can't grind up raw carbon, and hydrogen is unobtainable.
> 
> Also, minor thing, but masks of some kind would be cool.
> 
> Additional minor thing, naming your shank would be sick as fuck. Can't wait to see their final form. More insane feature idea: I wonder if you could use the cargo trolley linking system to "hook up" a primitive sled to a shank, have it haul your trade goods around.


Bug found: filled stone dropper is invisible.

**Comment by QRQ-2817** [comment]
Another bug: cooking spit does not change icons when doused.

**Comment by Captain277** [comment]
> Unsure on how to make them, I just wanted an option to not spawn with armor like I'm some experienced warrior.
> 
> Also, tribal-crafted items are supposed to be valuable to be tradable, right? Are the tools any different than a basic set? Could have them function quicker or something. Additionally, the primitive welder causes eye damage, while we have no eye protection. I attempted to make Imidazoline to fix my eyes, but can't grind up raw carbon, and hydrogen is unobtainable.
> 
> Also, minor thing, but masks of some kind would be cool.
> 
> Additional minor thing, naming your shank would be sick as fuck. Can't wait to see their final form. More insane feature idea: I wonder if you could use the cargo trolley linking system to "hook up" a primitive sled to a shank, have it haul your trade goods around.

Fixed the dropper bug, but I haven't been able to replicate the spit bug. As for tribal clothing, I see what you mean on that front. I'll see what I can do. The welder shouldn't be causing eye damage any more. I coded it to stop doing that, I thought.

**Comment by Captain277** [comment]
As a correction to the above, I have fixed both the welder and spit bugs. Thanks for reporting them!

**Review by silicons — APPROVED** [review]

**Pull request merged**